### PR TITLE
Added tests, data parser and data providers, created suite xml files …

### PR DESCRIPTION
**Pull request by tati1228**
…for smoke, negative and regression test.

To be executed using mvn test command.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            <artifactId>testng</artifactId>
+            <version>7.4.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.github.bonigarcia</groupId>
+            <artifactId>webdrivermanager</artifactId>
+            <version>5.2.0</version>
+        </dependency>
+        <dependency>
```
Comment (voropa) [inline pom.xml:28]: Why do you need this JUnit dependency, we are using TestNG instead

File: src/test/java/pages/ProductsPage.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public boolean isHeaderContainerDisplayed() {
+        return driver.findElement(headerContainer).isDisplayed();
+    }
+
+    public void clickAddRemoveButton(int productId) {
+        getAllProductContainers().get(productId).findElement(addRemoveButton).click();
+    }
+
+    public void addRemoveAllProducts() {
+        for (int i = 0; i < 6; i++) {
```
Comment (voropa) [inline src/test/java/pages/ProductsPage.java]: What if products count changes to any other number?
It's better to iterate through all actually avaliable products at the moment e.g.
for (product of getAllProductContainers()) {
product.findElement(addRemoveButton).click();
}

File: src/test/java/pages/ProductsPage.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     public void openProductDetails(String productName) {
+        WebElement productContainer = itemContainer(productName);
+        productContainer.findElement(itemDetailsLink).click();
+    }
+
+    public WebElement itemContainer(String productName) {
+        return driver.findElement(By.xpath(String.format(itemContainer, productName)));
+    }
+
+    private List<String> getExpectedOptionNames() {
```
Comment (voropa) [inline src/test/java/pages/ProductsPage.java]: Page object classes should not operate with any kind of Expected data. All expected test data should be stored in test classes / test data files

File: src/test/java/tests/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    @BeforeClass(alwaysRun = true)
+    public void setUp() {
+        String browser = System.getProperty("browser");
+        if (browser.equals("chrome")) {
+            WebDriverManager.chromedriver().setup();
+            ChromeOptions options = new ChromeOptions();
+            options.addArguments("headless");
+            driver = new ChromeDriver(options);
+         //   driver = new ChromeDriver();
```
Comment (voropa) [inline src/test/java/tests/BaseTest.java]: Please remove commented lines of code

File: src/test/java/pages/ProductsPage.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    public ProductsPage(WebDriver driver) {
+        super(driver);
+    }
+
+
+    public boolean isHeaderContainerDisplayed() {
+        return driver.findElement(headerContainer).isDisplayed();
+    }
+
+    public void clickAddRemoveButton(int productId) {
```
Comment (voropa) [inline src/test/java/pages/ProductsPage.java]: It's better to find a product by name, not by index

**Review by voropa — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (tati1228) [reply pom.xml:28]: Was inserted by mistake. Will be removed.

File: src/test/java/pages/ProductsPage.java (hunk 2 above)
Comment (tati1228) [reply src/test/java/pages/ProductsPage.java]: Ok, will update methods.

File: src/test/java/pages/ProductsPage.java (hunk 3 above)
Comment (tati1228) [reply src/test/java/pages/ProductsPage.java]: Ok, will move expected data to the test class.

File: src/test/java/tests/BaseTest.java (hunk 4 above)
Comment (tati1228) [reply src/test/java/tests/BaseTest.java]: Will do.

File: src/test/java/pages/ProductsPage.java (hunk 5 above)
Comment (tati1228) [reply src/test/java/pages/ProductsPage.java]: Will change tests to add a product by name

**Review by voropa — APPROVED** [review]

**Comment by tati1228** [comment]
Committed following changes:
Removed JUnit dependency;
Changed AddRemoveAllProducts method;
Test data is moved to the test method;
Removed commented code;
Changed method to add a product by name.

**Pull request merged**